rust/parquet/src/file/footer.rs

```diff
@@ -44,30 +44,31 @@ use crate::schema::types::{self, SchemaDescriptor};
 /// The reader first reads DEFAULT_FOOTER_SIZE bytes from the end of the file.
 /// If it is not enough according to the length indicated in the footer, it reads more bytes.
 pub fn parse_metadata<R: ChunkReader>(chunk_reader: &R) -> Result<ParquetMetaData> {
-    // check file is large enough to hold footer
-    let file_size = chunk_reader.len();
-    if file_size < (FOOTER_SIZE as u64) {
+    // read and cache up to DEFAULT_FOOTER_READ_SIZE bytes from the end and process the footer
+    let mut first_end_read = chunk_reader.get_read(
```

Comment:
It's weird that we are handling the "chunked read" logic in the footer reader - should this be inside the `ChunkReader` implementation? e.g., the reader itself should do lazy loading on the input stream based on how application decide to seek & read the data.

Reply:
That's a good point. I copied this logic from the C++ implem but it's not great in terms of seperation of concerns. If I understand you point: 
- here we should request a reader with `get_read(ChunkMode::FromEnd(FOOTER_SIZE))`
- it should be up to the `ChunkReader` implem to buffer extra bytes if it founds that `FOOTER_SIZE` is too small and it is cheap for him to get more bytes (and for instance the S3 ChunkReader will get 16kB instead)

Reply:
yes exactly, I feel the footer reader should not be aware of how the input stream is processed and also the logic can vary depending on the remote storage so the `DEFAULT_FOOTER_READ_SIZE` may not fit for all.